src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,210 @@
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonBar;
+import javafx.scene.control.ButtonBar.ButtonData;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.SelectionMode;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeTableColumn;
+import javafx.scene.control.TreeTableView;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.stage.Stage;
+
+public class KeyBindingsDialogViewModel {
+
+    private KeyBindingRepository keyBindingRepository;
+
+    @FXML
+    private TreeTableView<KeyBindingWrapper> keyBindingsTable;
+
+    @FXML
+    private TreeTableColumn<KeyBindingWrapper, String> actionColumn;
+
+    @FXML
+    private TreeTableColumn<KeyBindingWrapper, String> shortcutColumn;
+
+    @FXML
+    private Button closeButton;
+
+    @FXML
+    private Button resetButton;
+
+
+    @FXML
+    private void initialize() {
+        keyBindingsTable.getSelectionModel().setSelectionMode(SelectionMode.SINGLE);
+        ButtonBar.setButtonData(resetButton, ButtonData.LEFT);
+    }
+
+    public void initializeView() {
+        registerKeyEvents();
+        populateTable();
+        bindColumnValues();
+    }
+
+    /**
+     * Read all keybindings from the keybinding repository and create table keybinding
+     * models for them
+     */
+    private void populateTable() {
+        TreeItem<KeyBindingWrapper> root = new TreeItem<>(new KeyBindingWrapper(null, "", KeyBindingCategory.FILE));
+        for (KeyBindingCategory category : KeyBindingCategory.values()) {
+            TreeItem<KeyBindingWrapper> categoryItem = new TreeItem<>(new KeyBindingWrapper(null, "", category));
+            keyBindingRepository.getKeyBindings().forEach((keyBinding, bind) -> {
+                if (keyBinding.getCategory() == category) {
+                    KeyBindingWrapper keyBindWrapper = new KeyBindingWrapper(keyBinding, bind,
+                            keyBinding.getCategory());
+                    TreeItem<KeyBindingWrapper> keyBindTreeItem = new TreeItem<>(keyBindWrapper);
+                    categoryItem.getChildren().add(keyBindTreeItem);
+                }
+            });
+            categoryItem.setExpanded(true);
+            root.getChildren().add(categoryItem);
+        }
+        root.setExpanded(true);
+        keyBindingsTable.setRoot(root);
+    }
+
+    private void bindColumnValues() {
+        actionColumn
+                .setCellValueFactory(cellData -> new SimpleStringProperty(cellData.getValue().getValue().getName()));
+        shortcutColumn.setCellValueFactory(cellData -> cellData.getValue().getValue().bindProperty());
+    }
+
+    private void registerKeyEvents() {
+        keyBindingsTable.setOnKeyPressed(evt -> grabKey(evt));
+    }
+
+    private Stage getDialogStage() {
```

Comment:
Can be replaced by https://github.com/boceckts/jabref/blob/a559e058c065fe58cb5ced1380dda4f7ed889f51/src/main/java/net/sf/jabref/gui/FXAlert.java#L146 ?
In this case it may be worthwhile to add a closeDialog method to FXAlert.